carmm/run/workflows/helper.py

```diff
@@ -0,0 +1,94 @@
+import os
+from fnmatch import fnmatch
+from ase.io import read
+
+class CalculationHelper:
+    def __init__(self,
+                 calc_type,
+                 parent_dir,
+                 filename,
+                 restart=True,
+                 verbose=True):
+
+        self.calc_type = calc_type
+        self.parent_dir = parent_dir
+        self.filename = filename
+        self.restart = restart
+        self.verbose = verbose
+        self.counter = 0
+        self.prev_calcs = None
+        self.interpolation = None
+
+    def _find_restart(self):
+        restart_found = False
+        initial = None
+        while not restart_found and self.counter > 0:
+            subdirectory_name_previous = f"{self.calc_type}_{self.filename}_{self.counter - 1}"
+
+            # Handle different types of calculations
+            # Note that Vib does not need to be handled here
+            if self.calc_type == "Opt":
+                opt_restarts = 0
+                while os.path.exists(os.path.join(subdirectory_name_previous, f"{self.counter - 1}_{self.filename}_{opt_restarts}.traj")):
+                    opt_restarts += 1
+
+                traj_name = os.path.join(subdirectory_name_previous, f"{self.counter - 1}_{self.filename}_{opt_restarts - 1}.traj")
+
+                while os.path.exists(traj_name):
+                    if os.path.getsize(traj_name):
+                        if self.verbose:
+                            print(f"Restarting calculation from {traj_name}")
+                        initial = read(traj_name)
+                        restart_found = True
```

Comment:
Not sure, perhaps an extra boolean variable could be used, but it would lead to lesser clarity. The while loop is not uncontrolled here, it depends on the os.path.exists(), once the counter goes to below 0, the folders will not be found and the while loop will terminate anyway. Thoughts?

Reply:
I would be happy with a Boolean variable that is included in the `while` statement (i.e. `while (… and …)`). The problem with the break statement is if someone edits this in the future, it can become uncontrolled. So it is best practice to always use some kind of variable to control the exit. `break` should be avoided wherever possible - and we can avoid it here pretty easily.

Reply:
The `break` statement removed, the `restart_found` variable is reused.